File: 백준/Bronze/8958. OX퀴즈/OX퀴즈.cc
#include <bits/stdc++.h>
using namespace std;
string cs;
int N,i,cnt,sum;
int main(){
    ios_base::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    cin>>N;
    
    for(i=0; i<N; i++){
        cnt=0;
        sum=0;
        cin>>cs;
        for(char c : cs){
            if(c=='O'){
                cnt+=1;
                sum+=cnt;
            }
            else {
                cnt=0;
            }
        }
        cout<<sum<<"\n";
    }
    return 0;
}

File: 백준/Bronze/8958. OX퀴즈/OX퀴즈.py
n = int(input())

for _ in range(n):
    s = input()
    cnt = 0
    sum = 0
    for i in range(len(s)):
        if s[i] == "O":
            cnt += 1
            sum += cnt
        else:
            cnt = 0
    print(sum)